misk-cron/src/main/kotlin/misk/cron/CronTask.kt

```diff
@@ -0,0 +1,42 @@
+package misk.cron
+
+import com.google.common.util.concurrent.AbstractIdleService
+import misk.clustering.lease.LeaseManager
+import misk.logging.getLogger
+import misk.tasks.RepeatedTaskQueue
+import misk.tasks.Status
+import java.time.Duration
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+internal class CronTask @Inject constructor() : AbstractIdleService() {
+  @Inject private lateinit var cronClient: CronManager
+  @Inject private lateinit var leaseManager: LeaseManager
+  @Inject @ForMiskCron private lateinit var taskQueue: RepeatedTaskQueue
+
+  override fun startUp() {
+    logger.info { "Starting CronTask" }
+    val lease = leaseManager.requestLease(CLUSTER_LEASE_NAME)
+
+    taskQueue.scheduleWithBackoff(INTERVAL) {
+      if (lease.checkHeld()) {
+        cronClient.runReadyCrons()
+        logger.info { "Executed CronTask" }
+      }
+      Status.OK
+    }
+  }
+
+  override fun shutDown() {
+    logger.info { "Stopping CronTask" }
+    cronClient.removeAllCrons()
+  }
+
+  companion object {
+    private const val CLUSTER_LEASE_NAME = "misk.cron.crontask-cluster"
+    private val INTERVAL = Duration.ofSeconds(60L)
```

Comment:
Is it typical for a cron job to have an error of up to 60 seconds?

Reply:
I think for the granularity we're doing that would be fine.  I don't think we would have any tasks that would run every 60 seconds.